### PR TITLE
Fixes #848 add CLI mode to manually concatenate ts parts

### LINKED ISSUE #848: [Feature request] TwitchDownloaderCLI: add option to merge existing ts parts into custom output.ts
### Checklist

- [X] I have checked the [issue page](https://github.com/lay295/TwitchDownloader/issues?q=is%3Aissue) for duplicates
- [X] I am running the latest version ([download here](https://github.com/lay295/TwitchDownloader/releases/latest))

### Write your feature request here

What I am requesting is a way to directly call this method:
https://github.com/lay295/TwitchDownloader/blob/fefd3d18f84185977b9959032dd423b7e1fed405/TwitchDownloaderCore/VideoDownloader.cs#L663-L698

so I can merge already downloaded ts parts into custom `output.ts`, for example using ranges (like ffmpeg does) and individual files:
```
TwitchDownloaderCLI merge 10.ts -ss 14-muted.ts -to 17-muted.ts 20.ts -o output_custom.ts
```
So `custom.ts` will be made of (in the same order as given in the argument):
```
10.ts
14-muted.ts
15-muted.ts
16-muted.ts
17-muted.ts
20.ts
```

This custom way of calling CombineVideoParts call can help debug issues merging ts files with .NET Framework.

Also please add all the information to documentation.

### PR DESCRIPTION
**Pull request by superbonaci**

### CODE REVIEW

File: TwitchDownloaderCLI/README.md [hunk 1]
```diff
@@ -395,3 +413,9 @@ Default true boolean flags must be assigned: `--default-true-flag=false`. Defaul
 For Linux users, ensure both `fontconfig` and `libfontconfig1` are installed. `apt-get install fontconfig libfontconfig1` on Ubuntu.
 
 Some distros, like Linux Alpine, lack fonts for some languages (Arabic, Persian, Thai, etc.) If this is the case for you, install additional fonts families such as [Noto](https://fonts.google.com/noto/specimen/Noto+Sans) or check your distro's wiki page on fonts as it may have an install command for this specific scenario, such as the [Linux Alpine](https://wiki.alpinelinux.org/wiki/Fonts) font page.
+
+The list file for tsmerge can contain relative or absolute paths, but the path format differs from Windows to Linux/UNIX/MacOS (like volume paths or "/" instead of "\" to separate directories).
+The list must be a text file and describe one ts path per line.
+
+The concatenation made by tsmerge is not a raw (binary) concatenation, like `dd`, `cat` or `pv` would do on Linux, or `copy /b` on Windows.
+Instead, the streams have to be partially recoded, to keep its structure valid and playable. Using other software may not achieve this result.
```
Comment (ScrubN) [inline TwitchDownloaderCLI/README.md:421]: This function should be functionally equivalent to `cat`/`copy /b`, no? It quite literally opens a file, copies the bytes to a destination, then closes the file:
```cs
 await using (var fs = File.Open(partFile, FileMode.Open, FileAccess.Read, FileShare.Read))
{
    await fs.CopyToAsync(outputStream, cancellationToken).ConfigureAwait(false);
}
```

And I'm not sure why this says the ts files are partially recoded? We rely solely on FFmpeg for video encoding, and this function doesn't ever call FFmpeg.

**Comment by ScrubN** [comment]
When kick-support gets merged, it will bring an M3U8 parser with it (among several other changes).

Either I can refactor this feature (and its documentation) after that is merged, or I can merge this now and a temporary disclaimer is added that explains that there may be breaking changes in the next version (1.53.6). Which would you prefer?

File: TwitchDownloaderCLI/README.md (hunk 1 above)
Comment (superbonaci) [reply TwitchDownloaderCLI/README.md:421]: No it's not a raw concatenation, the resulting file has different size than the sum of its parts.

**Comment by superbonaci** [comment]
Merge it if you can, I think you can refactor later everything you need even add more features.

File: TwitchDownloaderCLI/README.md (hunk 1 above)
Comment (ScrubN) [reply TwitchDownloaderCLI/README.md:421]: Using `cat *.ts > cat.ts` under WSL I get the exact same result as I do from `CombineVideoParts`.
image
image

In order to achieve this result with cat, however, I had to pad the filenames with zeroes so they were enumerated in the correct order.

**Comment by ScrubN** [comment]
I decided to backport the M3U8 parser since kick-support still has some kinks to be ironed out. It should be pretty straight forward to use that instead of looping through a generic input file (and update the docs to reflect the change), however I am writing some extension methods right now to make it even easier for you.

**Comment by superbonaci** [comment]
Thanks for the merge. I can't confirm wether the `CopyToAsync` method does always perform a raw concatenation or not, I'll have to ask people with deeper knowledge than me. All the recent tests I did recently do it.

**Comment by ScrubN** [comment]
#917 has been merged, so you should be able to replace:
```cs
string InputList = downloadOptions.InputList;
List<string> videoPartsList = System.IO.File.ReadLines(InputList).ToList();
```
with:
```cs
await using var fs = File.Open(downloadOptions.InputList, FileMode.Open, FileAccess.Read, FileShare.Read);
var m3u8 = M3U8.FromStream(fs, Encoding.UTF8);
```

You may refer to `VideoDownloader.cs` on how to modify the other functions.
Don't forget to mention the user should input `.m3u`/`.m3u8` files in the documentation.
Also, consider renaming `downloadOptions` to better reflect what type of options it is.

**Comment by ScrubN** [comment]
I decided to make some of the changes myself, I hope you don't mind.

I also realized that limiting the user to M3U/M3U8 files was dumb, as it just requires the user to do more research before using the tool.

**Review by superbonaci — COMMENTED** [review]
review

**Comment by ScrubN** [comment]
Sorry this took so long to merge.

**Pull request merged**